syntax = "proto3";

package ark_nova_stats.bga_log_parser.proto;

import "ark_nova_stats/bga_log_parser/proto/action.proto";

message Round {
    // Corresponds to the actions taken up to a break.

    repeated Action actions = 1;
    // TODO: break rewards.
}

message GameAbandon {}

message GameTie {}

message GameEndResult {
    int32 rank = 1;
    int64 player_id = 2;
}

message GameEnd {
    GameEndResult result = 1;
}

message Game {
    int64 table_id = 1;

    oneof result {
        GameAbandon abandon = 2;
        GameTie tie = 3;
        GameEnd end = 4;
    }

    repeated int64 player_ids = 5;
    repeated Round rounds = 6;
}
